test/dev/ui/spinner_test.rb

```diff
@@ -21,6 +21,35 @@ def test_spinner
         )
       end
 
+      def test_updating_title
+        out, err = capture_io do
+          Dev::UI::StdoutRouter.ensure_activated
+          Dev::UI::Spinner.spin('watashi') do |task|
+            assert task
+            assert_respond_to task, :update_title
+            sleep Dev::UI::Spinner::PERIOD * 2.5
+            task.update_title 'kyou'
+            sleep Dev::UI::Spinner::PERIOD * 2.5
+            task.update_title 'tsukaretanda'
+            sleep Dev::UI::Spinner::PERIOD * 2.5
+          end
+        end
+
+        assert_empty err
+        match_lines(
+          out,
+          /⠋ watashi/,
+          /⠙/,
+          /⠹/,
+          /⠸ kyou/,
+          /⠼/,
+          /⠴ tsukaretanda/,
```

Comment:
If you're gonna put japanese in the tests, at least use hiragana/katakana/kanji instead of romaji